src/zcash/IncrementalMerkleTree.cpp

```diff
@@ -0,0 +1,287 @@
+#include <stdexcept>
+
+#include <boost/foreach.hpp>
+
+#include "zcash/IncrementalMerkleTree.hpp"
+#include "crypto/sha256.h"
+#include "zerocash/utils/util.h" // TODO: remove these utilities
+
+namespace libzcash {
+
+SHA256Compress SHA256Compress::combine(const SHA256Compress& a, const SHA256Compress& b)
```

Comment:
Could SHA256Compress be called SHA256CompressOutput instead? I misinterpreted it as though it was a thing that implements the SHA256 compression function (which made the code below confusing).